src/main/java/com/revo/deployr/client/util/RDataUtil.java

```diff
@@ -174,25 +174,35 @@ private static RData buildPrimitiveFromJSON(String name, Map json) {
         String rclass = (String) json.get("rclass");
 
         log.debug("buildPrimitiveFromJSON: name=" + name + " type=" + type + " rclass=" + rclass + " json=" + json);
-
+    
         if (json.containsKey("value")) {
 
             Object value = json.get("value");
 
             log.debug("buildPrimitiveFromJSON: name=" + name + " has value=" + value);
 
-            if (value instanceof Integer) {
-                rData = new RNumericImpl(name, ((Integer) value).doubleValue());
-            } else if (value instanceof Double) {
-                rData = new RNumericImpl(name, (Double) value);
-            } else if (value instanceof String) {
-                rData = new RStringImpl(name, (String) value);
-            } else if (value instanceof Boolean) {
-                rData = new RBooleanImpl(name, (Boolean) value);
-            } else {
-                // Indeterminate type, skip here to treat as RDataNA.
+            if("character".equalsIgnoreCase(rclass)){
+            	rData = new RStringImpl(name,  value == null ? null : (String) value);            	
+            }else if("numeric".equalsIgnoreCase(rclass)){
+            	rData = new RNumericImpl(name, value == null ? null : (Double) value);
+            }else if("logical".equalsIgnoreCase(rclass)){
+            	rData = new RBooleanImpl(name, value == null ? null :(Boolean) value);
             }
 
+            if(rData == null){
```

Comment:
For a fix that is just for null values, I would only expect a if(rData == null) {} clause added and nothing else changed in the rest of the code, to make sure we don't introduce any regression. I'm not sure why original lines 184-193 had to be replaced with new lines 184-189